# 2. 각 항목의 기여 정도:환적 편
전체 물동량을 수출, 수입, 환적 등의 요소로 나누어 각각이 전체 성장에 얼마나 기여 했는지를 살펴볼 수 있음.
예시)
전체 컨테이너 물동량 2000만 -> 2400만 증가
수출입 물동량 1000만 -> 1050만
환적 1000만 -> 1350만
== 전체 증가량의 대부분을 환적이 차지

1. 환적 물동량의 전반적인 증가량
2. 전체 물동량에서 환적이 차지하는 비중

In [2]:
import pandas as pd
import numpy as np

In [9]:
growth = pd.read_csv(
    r'C:\practice1\pr1\trans_table.csv',
    encoding='cp949',
    index_col='연도'
)

growth.head()

,수출입,환적,전체물동량,환적비율
연도,,,,
2001,5129694.00,2943107.75,8072801.75,36.46
2002,5566104.00,3887984.00,9454088.00,41.12
2003,6157128.25,4251414.25,10408542.50,40.85
2004,6698925.00,4791840.00,11490765.00,41.70
2005,6663670.75,5179724.00,11843394.75,43.74


첫 번째 ipynb
→ trans_table 또는 copy본 생성
→ trans_table.csv 저장

새 ipynb
→ trans_table.csv를 읽어서 growth 생성
→ growth만 수정

CSV 파일
→ 다시 저장하지 않는 한 그대로 유지

In [11]:
pd.options.display.float_format = '{:,.2f}'.format

growth.head()

,수출입,환적,전체물동량,환적비율
연도,,,,
2001,"5,129,694.00","2,943,107.75","8,072,801.75",36.46
2002,"5,566,104.00","3,887,984.00","9,454,088.00",41.12
2003,"6,157,128.25","4,251,414.25","10,408,542.50",40.85
2004,"6,698,925.00","4,791,840.00","11,490,765.00",41.70
2005,"6,663,670.75","5,179,724.00","11,843,394.75",43.74


In [12]:
growth['전체증감'] = growth['전체물동량'].diff()

growth['환적증감'] = growth['환적'].diff()

growth.head(10)

,수출입,환적,전체물동량,환적비율,전체증감,환적증감
연도,,,,,,
2001,"5,129,694.00","2,943,107.75","8,072,801.75",36.46,NaN,NaN
2002,"5,566,104.00","3,887,984.00","9,454,088.00",41.12,"1,381,286.25","944,876.25"
2003,"6,157,128.25","4,251,414.25","10,408,542.50",40.85,"954,454.50","363,430.25"
2004,"6,698,925.00","4,791,840.00","11,490,765.00",41.70,"1,082,222.50","540,425.75"
2005,"6,663,670.75","5,179,724.00","11,843,394.75",43.74,"352,629.75","387,884.00"
2006,"6,830,813.25","5,208,295.25","12,039,108.50",43.26,"195,713.75","28,571.25"
2007,"7,450,248.75","5,811,591.00","13,261,839.75",43.82,"1,222,731.25","603,295.75"
2008,"7,644,942.00","5,808,188.00","13,453,130.00",43.17,"191,290.25","-3,403.00"
2009,"6,607,973.75","5,372,636.25","11,980,610.00",44.84,"-1,472,520.00","-435,551.75"


In [14]:
growth['환적기여도'] = (
    growth['환적증감']
    / growth['전체증감']
    * 100
).round(2)

growth.head(25)

,수출입,환적,전체물동량,환적비율,전체증감,환적증감,환적기여도
연도,,,,,,,
2001,"5,129,694.00","2,943,107.75","8,072,801.75",36.46,NaN,NaN,NaN
2002,"5,566,104.00","3,887,984.00","9,454,088.00",41.12,"1,381,286.25","944,876.25",68.41
2003,"6,157,128.25","4,251,414.25","10,408,542.50",40.85,"954,454.50","363,430.25",38.08
2004,"6,698,925.00","4,791,840.00","11,490,765.00",41.70,"1,082,222.50","540,425.75",49.94
2005,"6,663,670.75","5,179,724.00","11,843,394.75",43.74,"352,629.75","387,884.00",110.00
2006,"6,830,813.25","5,208,295.25","12,039,108.50",43.26,"195,713.75","28,571.25",14.60
2007,"7,450,248.75","5,811,591.00","13,261,839.75",43.82,"1,222,731.25","603,295.75",49.34
2008,"7,644,942.00","5,808,188.00","13,453,130.00",43.17,"191,290.25","-3,403.00",-1.78
2009,"6,607,973.75","5,372,636.25","11,980,610.00",44.84,"-1,472,520.00","-435,551.75",29.58


| 전체증감 | 환적증감 | 의미                   |
| ---: | ---: | -------------------- |
|    + |    + | 환적이 전체 **성장에 기여**    |
|    + |    - | 환적이 전체 **성장을 억제**    |
|    - |    - | 환적이 전체 **감소에 기여**    |
|    - |    + | 환적 증가가 전체 **감소를 완화** |


In [ ]:
growth_year = growth[
    (growth['전체증감'] > 0) &
    (growth.index <= 2024)
].copy()

growth_year
# # 2005년 110%는 오류가 아니라, 수출입 물동량이 감소했음에도 환적 증가가 이를 상쇄했다는 해석

,수출입,환적,전체물동량,환적비율,전체증감,환적증감,환적기여도
연도,,,,,,,
2002,"5,566,104.00","3,887,984.00","9,454,088.00",41.12,"1,381,286.25","944,876.25",68.41
2003,"6,157,128.25","4,251,414.25","10,408,542.50",40.85,"954,454.50","363,430.25",38.08
2004,"6,698,925.00","4,791,840.00","11,490,765.00",41.70,"1,082,222.50","540,425.75",49.94
2005,"6,663,670.75","5,179,724.00","11,843,394.75",43.74,"352,629.75","387,884.00",110.00
2006,"6,830,813.25","5,208,295.25","12,039,108.50",43.26,"195,713.75","28,571.25",14.60
2007,"7,450,248.75","5,811,591.00","13,261,839.75",43.82,"1,222,731.25","603,295.75",49.34
2008,"7,644,942.00","5,808,188.00","13,453,130.00",43.17,"191,290.25","-3,403.00",-1.78
2010,"7,917,905.00","6,276,690.75","14,194,595.75",44.22,"2,213,985.75","904,054.50",40.83
2011,"8,832,312.00","7,352,641.75","16,184,953.75",45.43,"1,990,358.00","1,075,951.00",54.06


In [ ]:
high_contribution = growth_year[
    growth_year['환적기여도'] >= 50
].copy()

high_contribution

,수출입,환적,전체물동량,환적비율,전체증감,환적증감,환적기여도
연도,,,,,,,
2002,"5,566,104.00","3,887,984.00","9,454,088.00",41.12,"1,381,286.25","944,876.25",68.41
2005,"6,663,670.75","5,179,724.00","11,843,394.75",43.74,"352,629.75","387,884.00",110.00
2011,"8,832,312.00","7,352,641.75","16,184,953.75",45.43,"1,990,358.00","1,075,951.00",54.06
2012,"8,898,631.50","8,147,545.75","17,046,177.25",47.80,"861,223.50","794,904.00",92.30
2013,"8,937,645.50","8,748,453.00","17,686,098.50",49.47,"639,921.25","600,907.25",93.90
2014,"9,253,928.50","9,429,354.75","18,683,283.25",50.47,"997,184.75","680,901.75",68.28
2015,"9,363,406.25","10,105,318.25","19,468,724.50",51.91,"785,441.25","675,963.50",86.06
2018,"10,233,267.00","11,429,309.00","21,662,576.00",52.76,"1,169,100.75","1,150,902.00",98.44
2019,"10,354,104.75","11,637,895.50","21,992,000.25",52.92,"329,424.25","208,586.50",63.32


| 지표    | 질문                                |
| ----- | --------------------------------- |
| 환적비율  | 전체 물동량 중 환적이 얼마나 차지하는가?           |
| 환적기여도 | 전체 물동량이 증가한 양 중 환적 증가가 얼마나 차지했는가? |

In [ ]:
len(growth_year)

19

In [20]:
len(high_contribution)

11

In [21]:
len(high_contribution) / len(growth_year) * 100

57.89473684210527

In [22]:
round(
    len(high_contribution) / len(growth_year) * 100,
    2
)

57.89

`전체 물동량이 성장한 19개 연도 중 11개 연도(57.89%)에서 환적이 전체 증가분의 절반 이상을 차지했다`